stdlib/dynarray.mli

```diff
@@ -598,6 +598,18 @@ val reset : 'a t -> unit
     in the backing array at index [length a] or later.
 *)
 
+val unsafe_to_iarray : capacity:int -> ('a t -> unit) -> 'a iarray
+(** [unsafe_to_iarray ~capacity f] calls [f] on a new empty dynarray with the
+    given [capacity], then turns it into an immutable array without a copy,
+    when possible (i.e, the array is not a float array, or flat float arrays
+    are disabled).
+    When [f] returns, if the dynarray's capacity is not exactly equal to its
+    length, there will be a copy. Note that the dynarray's capacity is allowed
+    to change during the execution of [f].
```

Comment:
The writing of the comment could be restructured a bit, I think, because you detail some reasons why there could be a copy in two different paragraphs.

Reply:
It would be nice if the comment could make clear that the `capacity` argument is only a hint. For example, there is no copy if `[f]` calls `fit_capacity` on the dynarray, without any length change afterwards.